test/e2e/src/test-custom-types.js

```diff
@@ -144,6 +144,16 @@ test('u128', async t => {
   t.is((await t.context.client.u128({ u128: 1n })).result, 1n)
 })
 
+test('from', async (t) => {
+  function flattenInstance(object) {
+    return JSON.parse(JSON.stringify(object));
```

Comment:
In what way does this logic flatten an object? Removes dynamic fields?

Reply:
("flatten" to me means that it would turn things like `[ [1], [2, 3], 4]` into `[1, 2, 3, 4]`, but this doesn't do that)

Reply:
Its not really flattening it but it works around the fact that deepEquals requires the constructors to be the same, see https://github.com/avajs/ava/issues/1481#issuecomment-516502948

Reply:
Although I'm not 100% sure that it gives the exact equality we need.. which is why I also added the deepEquals for the ScSpecEntrys

Reply:
Marking this resolved for now, I updated the name to reflect what its actually doing.